### PR TITLE
VLM: GDAL Reproject and Resample RasterSources

### PR DESCRIPTION
**Pull request by pomadchin**
depends on #23, for a better diff, pr is against `feature/padded-tile` branch

### CODE REVIEW

File: vlm/src/test/scala/gdal/GDALReprojectRasterSourceSpec.scala [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            cols = bound.width, rows = bound.height)
+
+          val expected: Raster[MultibandTile] = {
+            val rr = implicitly[RasterRegionReproject[MultibandTile]]
+            rr.regionReproject(sourceTiff.raster, sourceTiff.crs, LatLng, testRasterExtent, testRasterExtent.extent.toPolygon, method)
+          }
+
+          val actual = warpRasterSource.read(bound).get
+
+          // actual.extent.covers(expected.extent) should be (true) -- doesn't work due to a precision issue
```
[the remark this answers is not in the document]
Comment (echeipesh) [reply vlm/src/test/scala/gdal/GDALReprojectRasterSourceSpec.scala]: Yeah, that's a good question. Floating point is going to happen so `+-` as a % of cell size seems safe. I'm a little concerned with the errors accumulating to grow the extent past the cell boundaries on the edge, picking up new rows/cols.

File: vlm/src/test/scala/gdal/GDALRasterSourceSpec.scala [hunk 2]
```diff
@@ -54,7 +54,7 @@ class GDALRasterSourceSpec extends FunSpec with RasterMatchers with BetterRaster
   }
 
   // no resampling is implemented there
-  ignore("should be able to resample") {
+  it("should be able to resample") {
```
[the remark this answers is not in the document]
Comment (echeipesh) [reply vlm/src/test/scala/gdal/GDALRasterSourceSpec.scala]: 🥇

**Comment by pomadchin** [comment]
Hm, @echeipesh mb we can merge it and to continue the work in a separate branch?

**Comment by pomadchin** [comment]
Merging it as it seems to me to be pretty clean enough for the master branch, will work on tests / features / bugs improvements in a separate PR.

**Pull request merged**